### DISCUSSION TITLE
Share URL: provide a way to manually copy the URL on iPad

### DISCUSSION THREAD

**Discussion by Stooovie** [navidrome/navidrome#3603] [Ideas]
On iPad, user is not allowed to get URL into clipboard automatically (`Error copying URL to clipboard: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permissio`n.) 

the error is displayed as a banner that auto-dismisses within seconds, not enough time to manually copy the link. This does not happen on the iPhone - there a popup is offered with a text field where the user can manually copy the URL. 

 I do access it via https, using Nginx Proxy Manager. 

Please provide a way to manually copy the shared URL on iPad as well. 

I have tried forcing the browser (Safari and Arc Search) to load mobile version of the Navidrome UI page, didn't work.

**Comment by metalheim** [comment]
Are you accessing navidrome via https?
If no, Browsers will Block certain features (like clipboard acccess)

If yes, then you probably just have to go into your Page settings and allow clipboard access for the navidrome URL. Not sure how this is called in Safari, but in Firefox and Chrome, you can set permissions for this per-URL

**Reply by Stooovie** [reply]
Yes I do. I added this to the post.

**Comment by kgarner7** [comment]
At least on iOS (pwa) clipboard access is restricted for me (over https). That being said, you can go to the share tab to copy the URL.

**Reply by Stooovie** [reply]
That's clunky as hell but thanks, will do in a pinch.

**Reply by radiosilence** [reply]
How do you get to the share tap? I'm using arc search on iOS and I can't seem to copy links (same error as OP).

I think better UX would be perhaps a modal with a text box and a copy link, but the ability to manually copy.

Could probably PR this but I am hungover as fuck

**Comment by Stooovie** [comment]
Just adding that various other self-hosted apps such as Gokapi or Immich allow copying URLs directly into clipboard without any issues. This seems to be a Navidrome issue, not an iOS limitation.
